src/components/Notifications/EventLog/usePrimaryToolbarFilterConfigWrapper.tsx

```diff
@@ -0,0 +1,234 @@
+            return createCustomFilterObj(bundleName, bundle, applications);
+        });
+
+        const completeBundleFilters = (filters.bundle as string[])?.map(queryParam => {
+            const bundle = bundles?.find(bundle => bundle.name === queryParam);
+            return createCustomFilterObj(queryParam, bundle, bundle?.children);
+        }) || [];
+
+        return completeBundleFilters.concat(partialBundleFilters) as EventLogCustomFilter[];
+    }, []);
+
+    // Run ONLY twice
+    //     1st - setup loading placeholders for filters
+    //     2nd - render actual filter bundles/applications
+    React.useEffect(() => {
+        setCustomFilters(mapToEventLogCustomFilter(filters, bundles.length !== 0 ? bundles : undefined));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [ bundles, mapToEventLogCustomFilter ]);
+
+    const applicationFilter = React.useMemo(() => {
+        return {
+            label: 'Application',
+            type: 'custom',
+            filterValues: {
+                children: <EventLogTreeFilter
+                    groups={ bundles }
+                    placeholder={ 'Filter by Application' }
+                    filters={ customFilters }
+                    updateFilters={ setCustomFilters }
+                />
+            }
+        } as any;
+    }, [ bundles, customFilters ]);
+
+    const activeFiltersConfig = React.useMemo(() => {
+        const activeFilters = toolbarConfig.activeFiltersConfig.filters as EventLogCustomFilter[];
+        const nonCustomFilters = activeFilters.filter(activeFilter => activeFilter && !activeFilter.bundleId);
+        return nonCustomFilters.concat(customFilters);
+    }, [ customFilters, toolbarConfig.activeFiltersConfig.filters ]);
+
+    // Update URL Query Params for Bundles
+    setFilters.bundle(produce(prev => {
+        if (bundles.length === 0) {
+            return filters.bundle;
+        }
+
+        const currBundleFilters: string[] = [];
+        bundles.forEach(bundle => {
+            const addToQueryParam = customFilters.some(bundleFilter => {
+                if (bundleFilter.bundleId === bundle.name) {
+                    // Edge case: Bundle has no children (but it gets a chip for UI reasons)
+                    if (bundle.children?.length === 0 && bundleFilter.chips.length === 1) {
+                        return true;
+                    }
+
+                    return bundle.children?.length === bundleFilter.chips.length;
+                }
+
+                return false;
+            });
+
+            if (addToQueryParam) {
+                currBundleFilters.push(bundle.name);
+            }
+        });
+
+        return areEqual(prev as string[], currBundleFilters, true) ? prev : currBundleFilters;
+    }));
+
+    // Update URL Query Params for Bundles
+    setFilters.application(produce(prev => {
```

Comment:
Is this executing on every call? I wonder if this should be inside a useEffect that depends on `setFilters.application` and `bundles`